pkg/crc/preflight/preflight_checks_linux.go

```diff
@@ -392,6 +395,54 @@ func fixLibvirtCrcNetworkActive() (bool, error) {
 	return true, nil
 }
 
+func checkLibvirtCrcNetworkPermissions() (bool, error) {
+	logging.Debug("Checking if 'crc' network has appropriate permissions setup")
+	path := qemuBridgeConfigPath1
+	_, err := os.Stat(path)
+	if err != nil {
+		logging.Debug("Failed to open %s: %s, trying %s..", qemuBridgeConfigPath1, err, qemuBridgeConfigPath2)
+		path = qemuBridgeConfigPath2
+		_, err := os.Stat(path)
+		if err != nil {
+			return false, fmt.Errorf("Error opening file: %s: %s", qemuBridgeConfigPath2, err.Error())
+		}
+	}
+	config, err := ioutil.ReadFile(filepath.Clean(path))
+	if err != nil {
+		return false, fmt.Errorf("Error opening file: %s: %s", path, err.Error())
+	}
+	if match, _ := regexp.MatchString(qemuBridgeConfig, string(config)); !match {
+		return false, fmt.Errorf("`crc` network not allowed unprivileged access")
+	}
+	logging.Debug("'crc' network has appropriate permissions")
+	return true, nil
+}
+
+func fixLibvirtCrcNetworkPermissions() (bool, error) {
+	logging.Debug("Fixing permissions for 'crc'")
+	path := qemuBridgeConfigPath1
+	_, err := os.Stat(path)
+	if err != nil {
+		logging.Debug("Failed to open %s: %s, trying %s..", qemuBridgeConfigPath1, err, qemuBridgeConfigPath2)
+		path = qemuBridgeConfigPath2
+		_, err := os.Stat(path)
+		if err != nil {
+			return false, fmt.Errorf("Error opening file: %s: %s", qemuBridgeConfigPath2, err.Error())
+		}
+	}
+	err = crcos.WriteToFileAsRoot(
```

Comment:
I might rather see an `AppendToFileAsRoot` as the meaning of `true` at the end is not clear until I scroll to this function.

You could keep the original function by name and parameters, extract the logic as-is now to a private/internal function which you call from either `Write` or `Append` with the appropriate bool value.